moped-editor/src/views/projects/projectView/ProjectComponents/index.js

```diff
@@ -85,12 +103,30 @@ export default function MapView({ projectName, projectStatuses }) {
 
   const [showEditModeDialog, setShowEditModeDialog] = useState(false);
 
+  const [addProjectComponent] = useMutation(ADD_PROJECT_COMPONENT);
+  const [deleteProjectComponent] = useMutation(DELETE_PROJECT_COMPONENT);
+  const { data, refetch: refetchProjectComponents } = useQuery(
+    GET_PROJECT_COMPONENTS,
+    {
+      variables: { projectId },
+      fetchPolicy: "no-cache",
+      onCompleted: () => {
+        setComponents(data.moped_proj_components);
+
+        // Create feature collections of all features in each component
+        const componentFeatureCollections = makeComponentFeatureCollectionsMap(
+          data.project_geography
+        );
+        setFeatureCollectionsByComponentId(componentFeatureCollections);
```

Comment:
Setting state here felt a little off to me at first, but it works well when `refetch()` is called. The alternative would be a useEffect that would watch `data` coming out of this `useQuery` or something like that.

Reply:
6 one way, half dozen the other. This doesn't feel wrong to me at all. The useEffect would be the way to go if there were other places in the app that manipulate the data coming out of the `useQuery()`, but I don't think that'll be the case here.

Reply:
Cool! I'm going to test this out before merging ✨

Reply:
I worked `useMemo` in, and I like it a lot since the previous state setter was only used once. TIL! 🙏